docs/frontend-system/building-plugins/testing.md

```diff
@@ -0,0 +1,237 @@
+  TestApiProvider,
+} from '@backstage/frontend-test-utils';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { CatalogApi, catalogApiRef } from '@backstage/plugin-catalog-react';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    const catalogApiMock: Partial<CatalogApi> = {
+      getEntityFacets: () =>
+        Promise.resolve({
+          facets: {
+            'relations.ownedBy': [{ count: 1, value: 'group:default/tools' }],
+          },
+        }),
+    };
+
+    const entityRef = stringifyEntityRef({
+      kind: 'Component',
+      namespace: 'default',
+      name: 'test',
+    });
+
+    renderInTestApp(
+      <TestApiProvider apis={[[catalogApiRef, catalogApiMock]]}>
+        <EntityDetails entitRef={entityRef} />
+      </TestApiProvider>,
+    );
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+## Testing features
+
+To facilitate testing of frontend features, the `@backstage/frontend-test-utils` package provides a tester class which starts up an entire frontend harness, complete with a number of default features. You can then provide overrides for extensions whose behavior you need to adjust for the test run.
+
+A number of features (frontend extensions and overrides) are also accepted by the tester. Here are some examples of how these facilities can be useful:
+
+### Single extension
+
+In order to test an extension in isolation, you simply need to pass it into the tester factory, then call the render method on the returned instance:
+
+```tsx
+import { screen } from '@testing-library/react';
+import { createExtensionTester } from '@backstage/frontend-test-utils';
+import { indexPageExtension } from './plugin';
+
+describe('Index page', () => {
+  it('should render a the index page', () => {
+    const tester = createExtensionTester(indexPageExtension);
+
+    tester.render();
+
+    expect(screen.getByText('Index Page')).toBeInTheDocument();
+  });
+});
+```
+
+### Extension preset
+
+There are some extensions that rely on other extensions existence, such as a page that links to another page. In that case, you can add more than one extension to the preset of features you want to render in the test, as shown below:
+
+```tsx
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { createExtensionTester } from '@backstage/frontend-test-utils';
+import { indexPageExtension, detailsPageExtension } from './plugin';
+
+describe('Index page', async () => {
+  it('should link to the details page', () => {
+    const tester = createExtensionTester(indexPageExtension);
+
+    // Adding more extensions to the preset being tested
+    tester.add(detailsPageExtension);
+    tester.render();
+
+    expect(screen.getByText('Index Page')).toBeInTheDocument();
```

Comment:
```suggestion
    await expect(screen.findByText('Index Page')).toBeInTheDocument();
```

it's just a neat habit to wait for results in case the minute details of the app makes them appear a single tick later or so

Reply:
Done 🙌🏻